pkg/project/expander.go

```diff
@@ -181,95 +185,178 @@ func expandPath(name string, script *Script) (string, error) {
 	return sf.Filename(), nil
 }
 
-// userExpander
-func userExpander(auth *authentication.Auth, element string) string {
-	if element == "name" {
-		return auth.WhoAmI()
+// ConstantExpander expands constants defined in the project-file.
+func ConstantExpander(_ string, name string, meta string, isFunction bool, ctx *Expansion) (string, error) {
+	projectFile := ctx.Project.Source()
+	constrained, err := constraints.FilterUnconstrained(pConditional, projectFile.Constants.AsConstrainedEntities())
+	if err != nil {
+		return "", err
 	}
-	if element == "email" {
-		return auth.Email()
+	for _, v := range constrained {
+		if v.ID() == name {
+			return projectfile.MakeConstantsFromConstrainedEntities([]projectfile.ConstrainedEntity{v})[0].Value, nil
+		}
 	}
-	if element == "jwt" {
-		return auth.BearerToken()
+	return "", nil
+}
+
+func TopLevelExpander(variable string, name string, _ string, _ bool, ctx *Expansion) (string, error) {
+	projectFile := ctx.Project.Source()
+	switch name {
+	case "project":
+		return projectFile.Project, nil
+	case "lock":
+		return projectFile.Lock, nil
```

Comment:
"project" shadows the field "Vars.Project", and "lock" (at the top level) will be difficult to keep updated using the introduced design. It would be nice to either drop these or include them within the Projvars struct.